halo2_proofs/src/plonk/circuit.rs

```diff
@@ -1508,6 +1623,31 @@ impl<F: Field, C: Into<Constraint<F>>, Iter: IntoIterator<Item = C>> IntoIterato
     }
 }
 
+/// GateV2Backend
+#[derive(Clone, Debug)]
+pub struct GateV2Backend<F: Field> {
```

Comment:
no reason other than using a placeholder suffix to avoid name collision :P  I agree that `v2` is not a great suffix

With the crate split we should get: `halo2_middleware::Gate` and `halo2_backend::Gate` or something similar.